packages/langium/src/serializer/json-serializer.ts

```diff
@@ -79,10 +91,39 @@ export class DefaultJsonSerializer implements JsonSerializer {
                     $error: value.error?.message ?? 'Could not resolve reference'
                 };
             }
+        } else if (textRegions && isAstNode(value)) {
+            const astNodeWithRegions = this.addAstNodeRegionWithAssignmentsTo(<AstNodeWithTextRegions>{ ...value });
+            if (value.$document && astNodeWithRegions?.$textRegion) {
+                astNodeWithRegions.$textRegion.documentURI = value.$document.uri.toString();
+            }
+            return astNodeWithRegions;
         }
         return value;
     }
 
+    protected addAstNodeRegionWithAssignmentsTo(node: AstNodeWithTextRegions) {
+        const createDocumentSegment: (cstNode: CstNode) => AstNodeRegionWithAssignments = cstNode => <DocumentSegment>{
+            offset: cstNode.offset,
+            end:    cstNode.end,
+            length: cstNode.length,
+            range:  cstNode.range,
+        };
+
+        if (node.$cstNode) {
+            const textRegion = node.$textRegion = createDocumentSegment(node.$cstNode);
+            const assignments: Record<string, DocumentSegment[]> = textRegion.assignments = {};
+
+            Object.keys(node).filter(key => !key.startsWith('$')).forEach( key => {
+                const propertyAssignments = findNodesForProperty(node.$cstNode, key).map( createDocumentSegment );
+                if (propertyAssignments.length !== 0)
```

Comment:
I'd add curly braces even to single lines for consistency.

Reply:
👍